src/main/java/edu/bu/metcs673/trackr/service/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package edu.bu.metcs673.trackr.service.impl;
+
+import edu.bu.metcs673.trackr.api.TransactionDTO;
+import edu.bu.metcs673.trackr.common.CommonConstants;
+import edu.bu.metcs673.trackr.common.TrackrInputValidationException;
+import edu.bu.metcs673.trackr.domain.BankAccount;
+import edu.bu.metcs673.trackr.domain.Transaction;
+import edu.bu.metcs673.trackr.repo.TransactionRepository;
+import edu.bu.metcs673.trackr.service.TransactionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class TransactionServiceImpl implements TransactionService {
+
+    @Autowired
+    private TransactionRepository transactionRepository;
+
+
+    @Override
+    public Transaction createTransaction(TransactionDTO transactionInput, BankAccount bankAccount) {
+        // create new Transaction object using DTO fields, set bank account as well
+        Transaction transaction = new Transaction(transactionInput, bankAccount);
+
+        return transactionRepository.save(transaction);
+    }
+
+    @Override
+    public Transaction findTraByIdAndBankAccountId(long transactionId, long bankAccountId) {
+        Transaction transaction;
+        if ((transaction = transactionRepository.findTraByIdAndBankAccountId(transactionId, bankAccountId)) != null) {
```

Comment:
This line is a little confusing.  Can you set `transaction` above the if-statement and then do the null check as the if-condition?

Reply:
OK, have done